src/core/cgroup.c

```diff
@@ -1062,15 +1063,30 @@ static void cgroup_context_apply(
 
 CGroupMask cgroup_context_get_mask(CGroupContext *c) {
         CGroupMask mask = 0;
+        bool need_cpumask_for_cpu_accounting;
+        struct utsname u;
+
+        /* In the unified cgroup hierarachy CPU accounting information is
+         * read from cpu.stat file. On Linux 4.15 and newer cpu.stat exists
+         * without CPU controller enabled. On the other hand, on older kernels
+         * CPU controller has to be enabled for cpu.stat to appear. Let's check it.
+         * Related discussion found at PR #9665. */
+        assert_se(uname(&u) >= 0);
+        need_cpumask_for_cpu_accounting = str_verscmp(u.release, "4.15") < 0;
 
         /* Figure out which controllers we need */
 
-        if (c->cpu_accounting ||
-            cgroup_context_has_cpu_weight(c) ||
+        if (cgroup_context_has_cpu_weight(c) ||
             cgroup_context_has_cpu_shares(c) ||
             c->cpu_quota_per_sec_usec != USEC_INFINITY)
                 mask |= CGROUP_MASK_CPUACCT | CGROUP_MASK_CPU;
 
+        if (c->cpu_accounting) {
+                mask |= CGROUP_MASK_CPUACCT;
+                if(need_cpumask_for_cpu_accounting)
+                        mask |= CGROUP_MASK_CPU;
```

Comment:
hmm, on cgroupsv1 we should still add CPUACCT and CPU into the mask here as before, after all we always mount both controllers together there.